providers/launchdarkly/pkg/provider.go

```diff
@@ -0,0 +1,366 @@
+	if ok && strings.Trim(k, " ") != "" {
+		kind = ldcontext.Kind(k)
+	} else {
+		p.l.Warn("no context kind set, setting %q by default", kind)
+	}
+
+	if kind == ldcontext.MultiKind {
+		p.l.Debug("multi context detected")
+		return p.toMultiLDContext(evalCtx)
+	}
+
+	p.l.Debug("single context detected")
+	return p.mapContext(kind, evalCtx)
+}
+
+// toReason maps LaunchDarkly flag evaluation reasons to OpenFeatures'
+func (p *Provider) toReason(reasonKind ldreason.EvalReasonKind) openfeature.Reason {
+	switch reasonKind {
+	case ldreason.EvalReasonOff:
+		return openfeature.DisabledReason
+	case ldreason.EvalReasonTargetMatch:
+		return openfeature.TargetingMatchReason
+	case ldreason.EvalReasonError:
+		return openfeature.ErrorReason
+	case ldreason.EvalReasonRuleMatch,
+		ldreason.EvalReasonPrerequisiteFailed,
+		ldreason.EvalReasonFallthrough:
+		fallthrough
+	default:
+		return openfeature.Reason(reasonKind)
+	}
+}
+
+// toResolutionError maps LaunchDarkly flag resolution errors to OpenFeatures'
+func (p *Provider) toResolutionError(errorKind ldreason.EvalErrorKind, reason string) openfeature.ResolutionError {
+	switch errorKind {
+	case ldreason.EvalErrorClientNotReady:
+		return openfeature.NewProviderNotReadyResolutionError(reason)
+	case ldreason.EvalErrorFlagNotFound:
+		return openfeature.NewFlagNotFoundResolutionError(reason)
+	case ldreason.EvalErrorMalformedFlag:
+		return openfeature.NewParseErrorResolutionError(reason)
+	case ldreason.EvalErrorUserNotSpecified:
+		return openfeature.NewTargetingKeyMissingResolutionError(reason)
+	case ldreason.EvalErrorWrongType:
+		return openfeature.NewTypeMismatchResolutionError(reason)
+	default:
+		return openfeature.NewGeneralResolutionError(reason)
+	}
+}
+
+// toProviderResolutionDetail maps LaunchDarkly's flag resolution details to
+// OPen
+func (p *Provider) toProviderResolutionDetail(detail ldreason.EvaluationDetail) openfeature.ProviderResolutionDetail {
+	p.l.Debug("launchdarkly evaluation detail: %v", detail)
+
+	ofDetail := openfeature.ProviderResolutionDetail{
+		Reason: p.toReason(detail.Reason.GetKind()),
+	}
+
+	if errorKind := detail.Reason.GetErrorKind(); errorKind != "" {
+		ofDetail.ResolutionError = p.toResolutionError(errorKind, fmt.Sprintf("LaunchDarkly returned %s", detail.Reason))
+	}
+
+	if detail.VariationIndex.IsDefined() {
+		ofDetail.Variant = detail.VariationIndex.String()
+	}
+
+	return ofDetail
+}
+
+// resolveContext encapsulates common logic that validates the OpenFeature evaluation
+// contest and translates it to LaunchDarkly's. Callers are supposed to only use
+// ldCtx if an error is not returned. When an error is returned, a non-empty OpenFeature
+// ResolutionDetail is returned as well with additional details.
+// This function also handles context cancellations.
+func (p *Provider) resolveContext(ctx context.Context, evalCtx openfeature.FlattenedContext) (ldcontext.Context, openfeature.InterfaceResolutionDetail, error) {
```

Comment:
[`Resolution`](https://openfeature.dev/specification/glossary#resolving-flag-values) is the term we use to mean "get flag value from provider". I think `tranformContext` might be a bit better a name for this func.

Reply:
ah, that makes sense! Thank @toddbaert!